src/main/scala/com.snowplowanalytics/weather/providers/openweather/OwmCacheClient.scala

```diff
@@ -42,16 +49,20 @@ import WeatherCache.{CacheKey, Position}
  *                     values 46.0, 45.5, 45.7, 45.78 by geoPrecision 1,2,10,100 respectively
  *                     geoPrecision 1 will give ~60km infelicity if worst case; 2 ~30km etc
  * @param asyncClient instance of `OwmAsyncClient` which will do all underlying work
- * @param timeout timeout in seconds after which active request will be considered failed
+ * @param requestTimeout timeout after which active request will be considered failed
  */
-class OwmCacheClient(val cacheSize: Int, val geoPrecision: Int, asyncClient: OwmAsyncClient, val timeout: Int)
-    extends Client[ValidatedWeather]
+class OwmCacheClient[F[_]: Concurrent](
```

Comment:
could we have the effect constraints at the method level as I suspect we don't need `Concurrent` everywhere

Reply:
``` scala
private val timer: Timer[F] = Timer.derive[F]
```
needs top-level `LiftIO[F]`, which is not too far from `Concurrent[F]` - so we will need a top-level context bound anyway. Do you think it's still worth it?

Reply:
yeah in that case no :+1: